client/fs_endpoint.go

```diff
@@ -473,7 +476,12 @@ OUTER:
 	}
 
 	if streamErr != nil {
-		f.handleStreamResultError(streamErr, helper.Int64ToPtr(500), encoder)
+		// If error has a Code, use it
+		var code int64 = 500
+		if codedErr, ok := streamErr.(interface{ Code() int }); ok {
```

Comment:
Did not realize you could do that

Reply:
It is not something I normally turn to, but in cases like this I think it's a reasonable shortcut as we can't reference the HTTPCodedError in `agent` (circular imports). I don't mind creating a named interface or something though.